src/table/components/virtualized-table/Body.tsx

```diff
@@ -32,69 +24,43 @@ const Body = (props: BodyProps) => {
     selectionsAPI,
     totals,
   } = props;
-  const { rowsInPage, loadData, debouncedLoadData } = useInfiniteScrollData(model, layout, pageInfo, columns);
-  const rowCount = Math.min(pageInfo.rowsPerPage, layout.qHyperCube.qSize.qcy - pageInfo.page * pageInfo.rowsPerPage);
-  const visibleRowCount = Math.min(rowCount, Math.ceil(rect.height / DEFAULT_ROW_HEIGHT));
-  const visibleColumnCount = useMemo(
-    () =>
-      columnWidth.reduce(
-        (data, colWidth) => {
-          if (data.width < rect.width) {
-            data.width += colWidth;
-            data.count += 1;
-          }
-
-          return data;
-        },
-        { count: 0, width: 0 }
-      ),
-    [rect, columnWidth]
-  ).count;
+  const { scrollHandler, verticalScrollDirection, horizontalScrollDirection } = useScrollDirection();
+  const { rowCount, visibleRowCount, visibleColumnCount } = useTableCount(layout, pageInfo, rect, columnWidth);
   const stickyTop = totals.atTop ? HEADER_AND_TOTALS_HEIGHT : HEADER_HEIGHT;
   let { height: bodyHeight } = rect;
   bodyHeight -= paginationNeeded ? PAGINATION_HEIGHT : 0;
   bodyHeight -= totals.shrinkBodyHeightBy;
   bodyHeight = Math.min(rowCount * DEFAULT_ROW_HEIGHT, bodyHeight);
 
-  useSelectionsEffect(selectionsAPI, rowsInPage);
+  const { rowsInPage, loadRows, loadColumns } = useData(
+    model,
+    layout,
+    pageInfo,
+    visibleRowCount,
+    visibleColumnCount,
+    columns
+  );
+
+  const handleItemsRendered = useItemsRendererHandler({
+    layout,
+    loadRows,
+    loadColumns,
+    verticalScrollDirection,
+    horizontalScrollDirection,
+    rowCount,
+    pageInfo,
+  });
 
-  useEffect(() => {
-    // Initial data load
-    const top = pageInfo.page * pageInfo.rowsPerPage;
-    loadData(0, top, visibleColumnCount, visibleRowCount);
-  }, [layout, visibleRowCount, visibleColumnCount, loadData, pageInfo]);
+  const itemData = useMemo(() => ({ rowsInPage, columns, bodyStyle }), [rowsInPage, columns, bodyStyle]);
```

Comment:
does this make a difference? Something about `VariableSizeGrid` not rerendering if `itemData` is unchanged?

Reply:
Have a vague memory of it being required to make the react-window `areEqual` function work properly. No I cannot find it any more 😕